lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -16,15 +16,16 @@ module Common
     VERSION_TYPES_PERMITTING_CONFLICT = ["external", "external_gt", "external_gte"]
 
     def register
+      @template_installed = Concurrent::AtomicBoolean.new(false)
       @stopping = Concurrent::AtomicBoolean.new(false)
       # To support BWC, we check if DLQ exists in core (< 5.4). If it doesn't, we use nil to resort to previous behavior.
       @dlq_writer = dlq_enabled? ? execution_context.dlq_writer : nil
 
       setup_hosts # properly sets @hosts
       build_client
-
-      install_template
+      after_successful_connection { install_template }
       check_action_validity
+      #after_successful_connection { install_template }
```

Comment:
this leftover commented code can be removed

Reply:
I thought I'd already pushed the commit that did this...